src/components/Exchange/SlippageInput.tsx

```diff
@@ -0,0 +1,91 @@
+import "./SlippageInput.scss";
+import { useState } from "react";
+import cx from "classnames";
+import { formatAmount } from "lib/numbers";
+
+const MAX_SLIPPAGE = 99 * 100;
+const HIGH_SLIPPAGE = 2 * 100;
+const SLIPPAGE_SUGGESTION_LISTS = [0.3, 0.5, 1, 1.5];
+const validDecimalRegex = /^\d+(\.\d{0,2})?$/; // 0.00 ~ 99.99
+
+function getSlippageText(value: number) {
+  return formatAmount(value, 2, 2).replace(/0+$/, "");
+}
+
+type Props = {
+  setAllowedSlippage: (value: number) => void;
+  defaultSlippage: number;
+};
+
+export default function SlippageInput({ setAllowedSlippage, defaultSlippage }: Props) {
+  const defaultSlippageText = getSlippageText(defaultSlippage);
+  const [slippageText, setSlippageText] = useState<string>(defaultSlippageText);
+  const [slippageError, setSlippageError] = useState<string>("");
+  const [isPanelVisible, setIsPanelVisible] = useState<boolean>(false);
+
+  function handleChange(event: React.ChangeEvent<HTMLInputElement>) {
+    setSlippageError("");
+    const { value } = event.target;
+    if (value === "") {
+      setSlippageText(value);
+      setAllowedSlippage(defaultSlippage);
+      return;
+    }
+
+    const parsedValue = Math.round(Number.parseFloat(value) * 100);
+    if (Number.isNaN(parsedValue)) {
+      return;
+    }
+
+    if (parsedValue >= HIGH_SLIPPAGE) {
```

Comment:
i think setState functions should be wrapped to useEffect to avoid endless updates just in case